src/genie/libs/parser/hp_comware/show_interface.py

```diff
@@ -0,0 +1,703 @@
+                interface_dict[interface]['counters']["in_parity_errors"] = int(in_parity_errors)
+                continue
+
+            # r'^ *Output: *(?P<out_errors>.*) output errors, (?P<out_underruns>.*) underruns, (?P<out_buffer_failure>.*) buffer failures$'
+            m = p26.match(line)
+            if m:
+                out_errors = m.groupdict()['out_errors']
+                out_underruns = m.groupdict()['out_underruns']
+                out_buffer_failure = m.groupdict()['out_buffer_failure']
+
+                if(out_errors == '-'):
+                    out_errors = 0
+                if(out_underruns == '-'):
+                    out_underruns = 0
+                if(out_buffer_failure == '-'):
+                    out_buffer_failure = 0
+
+                interface_dict[interface]['counters']["out_errors"] = int(out_errors)
+                interface_dict[interface]['counters']["out_underruns"] = int(out_underruns)
+                interface_dict[interface]['counters']["out_buffer_failure"] = int(out_buffer_failure)
+                continue
+
+            # ^ *(?P<out_abort>.*) aborts, (?P<out_deferred>.*) deferred, (?P<out_collision>.*) collisions, (?P<out_late_collision>.*) late collisions$
+            m = p27.match(line)
+            if m:
+                out_abort = m.groupdict()['out_abort']
+                out_deferred = m.groupdict()['out_deferred']
+                out_collision = m.groupdict()['out_collision']
+                out_late_collision = m.groupdict()['out_late_collision']
+
+                if(out_abort == '-'):
+                    out_abort = 0
+                if(out_deferred == '-'):
+                    out_deferred = 0
+                if(out_collision == '-'):
+                    out_collision = 0
+                if(out_late_collision == '-'):
+                    out_late_collision = 0
+
+                interface_dict[interface]['counters']["out_abort"] = int(out_abort)
+                interface_dict[interface]['counters']["out_deferred"] = int(out_deferred)
+                interface_dict[interface]['counters']["out_collision"] = int(out_collision)
+                interface_dict[interface]['counters']["out_late_collision"] = int(out_late_collision)
+                continue
+
+            # '^ *(?P<out_lost_carrier>.*) lost carrier, (?P<out_no_carrier>.*) no carrier$'
+            m = p28.match(line)
+            if m:
+                out_lost_carrier = m.groupdict()['out_lost_carrier']
+                out_no_carrier = m.groupdict()['out_no_carrier']
+
+                if(out_lost_carrier == '-'):
+                    out_lost_carrier = 0
+                if(out_no_carrier == '-'):
+                    out_no_carrier = 0
+
+                interface_dict[interface]['counters']["out_lost_carrier"] = int(out_lost_carrier)
+                interface_dict[interface]['counters']["out_no_carrier"] = int(out_no_carrier)
+                continue
+
+        return(interface_dict)
```

Comment:
no need for brackets I believe;
`return interface_dict`

Reply:
Fixed